crates/bevy_state/src/app.rs

```diff
@@ -120,6 +132,14 @@ impl AppExtStates for SubApp {
 
         self
     }
+
+    #[cfg(feature = "bevy_hierarchy")]
+    fn enable_state_scoped_entities<S: States>(&mut self) -> &mut Self {
+        self.add_systems(
+            StateTransition,
+            clear_state_scoped_entities::<S>.in_set(StateTransitionSteps::ExitSchedules),
```

Comment:
Is there a reason we can't use the `OnExit` schedules for this? Seems a bit unnecessary to always run it even when no changes happened, tho probably not blocking :thinking:

Reply:
This is actually the exact same way `OnExit` is executed, same schedule and same set.
I don't have a strong opinion on this, both will work exactly the same way, so if we want this with `OnExit` that's fine

Reply:
I'd like this in the `OnExit` schedules so users can order this relative to other systems.

Reply:
I remember now, `OnExit(value)` requires the `value`, meaning we'd need a different system for each possible state variant. Going one step back allows us to group all exit transitions, no matter which state we end up in.

I'll consider this resolved since the other approach is basically invalid